src/components/Discussion/index.jsx

```diff
@@ -0,0 +1,107 @@
+import { useState, useRef } from "react";
+import "./index.css";
+import Navbar from "../Navbar";
+import StarRating from "./rating";
+import Card from "./card";
+
+export default function Discussion() {
+  const [comment, setComment] = useState("");
+  const [name, setName] = useState("");
+  const [count, setCount] = useState(0);
+  const [rating, setRating] = useState(0);
+  const [commentList, setCommentList] = useState([]);
+  const commentRef = useRef(null);
+  const onClickHandler = (e) => {
+    e.preventDefault();
+    setName("");
+    commentRef.current.value = "";
+    if (name === "") {
+      window.alert("Enter the credentials");
+    } else {
+      setCommentList([
+        ...commentList,
```

Comment:
instead of directly using `commentList` you should access previous state using function and the update it